### DISCUSSION TITLE
How do I change behavior of completions to prepend ^ to special characters?

### DISCUSSION THREAD

**Discussion by digitalguy99** [chrisant996/clink#705] [Q&A]
Currently, the behavior of my completions(?) is to add double quotes to strings that contain special characters like spaces or brackets as shown in the following video:

https://github.com/user-attachments/assets/418e00c9-96be-4e9f-8c74-b9ea68e0fa0a

But I want it to prepend ^ instead of adding double quotes. So,
```
Program Files -> Program^ Files✅
Program Files -> "Program Files"❌
```
or
```
Program Files (x86) -> Program^ Files^ ^(x86^)✅
Program Files (x86) -> "Program Files (x86)"❌
```

Wondering how I can do this.

**Comment by chrisant996** [comment]
I think you're expecting that cmd.exe's `^` character works the same as bash's `\` character for escaping spaces. 

But it doesn't.  There's no way to make `^` be able to escape a space in cmd.exe.

This is just how cmd.exe works.  Clink can't do anything about that.

That's why completions with spaces have to use quotes.  There's simply no other way.

**Reply by chrisant996** [reply]
Actually it's not even about cmd.exe -- this is simply how the Windows OS works.

Because in Windows programs receive the command line as a single string, and the programs have to parse the command line to split it into separate arguments.  The `^` character in cmd.exe only affects cmd.exe's internal parsing, and it is stripped by cmd.exe before the command line reaches the spawned program.

So quotes are required in order for programs to know which spaces separate arguments, versus which spaces are embedded within an argument. 

Typically a programming language's runtime library takes care of parsing the command line into arguments.
